libs/parallelism/algorithms/include/hpx/parallel/algorithms/rotate.hpp

```diff
@@ -308,3 +395,120 @@ namespace hpx { namespace parallel { inline namespace v1 {
+            )>
+        // clang-format on
+        friend FwdIter tag_fallback_dispatch(
+            hpx::rotate_t, FwdIter first, FwdIter new_first, FwdIter last)
+        {
+            static_assert(hpx::traits::is_forward_iterator<FwdIter>::value,
+                "Requires at least forward iterator.");
+
+            return hpx::parallel::v1::detail::rotate<FwdIter>().call2(
+                hpx::execution::sequenced_policy{}, first, new_first, last);
+
+        }
+        // clang-format off
+        template <typename ExPolicy, typename FwdIter,
+            HPX_CONCEPT_REQUIRES_(
+                hpx::is_execution_policy<ExPolicy>::value &&
+                hpx::traits::is_forward_iterator<FwdIter>::value
+            )>
+        // clang-format on
+        friend typename parallel::util::detail::algorithm_result<ExPolicy, FwdIter>::type
+        tag_fallback_dispatch(
+            hpx::rotate_t, ExPolicy&& policy, FwdIter first, FwdIter new_first, 
+                FwdIter last)
+        {
+            static_assert((hpx::traits::is_forward_iterator<FwdIter>::value),
+                "Requires at least forward iterator.");
+
+            typedef std::integral_constant<bool,
+                hpx::is_sequenced_execution_policy<ExPolicy>::value ||
+                    !hpx::traits::is_bidirectional_iterator<FwdIter>::value>
+                is_seq;
+
+            return parallel::util::get_second_element(
+                hpx::parallel::v1::detail::rotate<hpx::parallel::util::in_out_result<FwdIter,FwdIter>>()
+                .call2(std::forward<ExPolicy>(policy), is_seq(), first, new_first, last));
+
+        } 
+
+    } rotate{};
+
+    ///////////////////////////////////////////////////////////////////////////
+    // CPO for hpx::rotate_copy
+    HPX_INLINE_CONSTEXPR_VARIABLE struct rotate_copy_t final
+      : hpx::functional::tag_fallback<rotate_copy_t>
+    {
+        // clang-format off
+        template <typename FwdIter, typename OutIter,
+            HPX_CONCEPT_REQUIRES_(
+                hpx::traits::is_iterator<FwdIter>::value&&    
+                hpx::traits::is_iterator<OutIter>::value
+            )>
+        // clang-format on
+        friend OutIter tag_fallback_dispatch(
+            hpx::rotate_copy_t, FwdIter first, FwdIter new_first, FwdIter last, 
+                OutIter dest_first)
+        {
+            static_assert((hpx::traits::is_forward_iterator<FwdIter>::value),
+                "Requires at least forward iterator.");
+            static_assert((hpx::traits::is_output_iterator<OutIter>::value),
+                "Requires at least output iterator.");
+
+            //typedef hpx::is_sequenced_execution_policy<ExPolicy> is_seq;
+
+            return parallel::util::get_second_element(parallel::v1::detail::rotate_copy<
+                hpx::parallel::util::in_out_result<FwdIter, OutIter>>()
+                .call2(hpx::execution::sequenced_policy{}, first, 
```

Comment:
```suggestion
                .call(hpx::execution::seq, first, 
```
I believe you will have to add tests for this algorithm overload. Such tests would have caught this.